glrd/query.py

```
@@ -277,26 +308,171 @@ def parse_arguments():
+        regular_flavors = {}
+        oci_flavors = {}
+        
+        for flavor in sorted(release['flavors']):  # Sort flavors for consistent ordering
+            # Get platform from flavor (first part before dash)
+            platform = flavor.split('-')[0]
+
+            # Handle container platform differently
+            if platform == "container":
+                # Remove the architecture part (last component after dash)
+                flavor_base = '-'.join(flavor.split('-')[:-1])
+                oci_flavors[flavor] = {
+                    'oci': f"{DEFAULTS['CONTAINER_REGISTRY']}:{version}"
+                }
+                continue
+            
+            # Handle bare platform differently
+            if platform == "bare":
+                # Remove the architecture part (last component after dash)
+                flavor_base = '-'.join(flavor.split('-')[:-1])
+                oci_flavors[flavor] = {
+                    'oci': f"{DEFAULTS['CONTAINER_REGISTRY']}/{flavor_base}:{version}"
+                }
+                continue
+            
+            # For all other platforms, use metadata and image URLs
+            # Get file extension for this platform
+            image_ext = DEFAULTS['PLATFORM_EXTENSIONS'].get(platform, "raw")
+            
+            base_url = f"{DEFAULTS['ARTIFACTS_S3_BASE_URL']}/{DEFAULTS['ARTIFACTS_S3_PREFIX']}{flavor}-{version}-{commit_short}"
+            base_filename = f"{flavor}-{version}-{commit_short}"
+            
+            regular_flavors[flavor] = {
+                'metadata': f"{base_url}/{base_filename}.manifest",
+                'image': f"{base_url}/{base_filename}.{image_ext}"
+            }
+        
+        # Combine regular and OCI flavors, with OCI flavors at the end
+        return {**regular_flavors, **oci_flavors}
+        
+    except Exception as e:
+        logging.error(f"Error formatting flavors with URLs: {e}")
+        return {}
```

Does it really make sense to hide exceptions and return an empty `dict` instead? If an exception occurs something is badly broken anyway.

---

What kind of Exceptions do we expect here? 

In case there is an **expected** exception, that we use here to implicitly check if we need to return an empty dict => we should make that check explicit if possible. 

Could not identify such a case yet, so I would say let it crash.

---

You are right. Returning an empty dict is only needed if (and already implemented if nothing else goes wrong).
I reworked all error codes and exceptions.